pkgs/development/python-modules/pytest-flake8/default.nix

```diff
@@ -0,0 +1,25 @@
+{lib, buildPythonPackage, fetchPypi, pytest, flake8}:
+
+buildPythonPackage rec {
+  name = "${pname}-${version}";
+  pname = "pytest-flake8";
+  version = "0.8.1";
+
+  propagatedBuildInputs = [ pytest flake8 ];
```

Comment:
do not add `pytest` as `propagatedBuildInput` but as `buildInput`. This package is an exception.